packages/@aws-cdk/pipelines/test/builds.test.ts

```diff
@@ -219,6 +220,75 @@ test('Standard (NPM) synth can output additional artifacts', () => {
   });
 });
 
+test('Pipeline action contains a hash that changes as the buildspec changes', () => {
+  const hash1 = synthWithAction((sa, cxa) => cdkp.SimpleSynthAction.standardNpmSynth({
+    sourceArtifact: sa,
+    cloudAssemblyArtifact: cxa,
+  }));
+  const hash2 = synthWithAction((sa, cxa) => cdkp.SimpleSynthAction.standardNpmSynth({
+    sourceArtifact: sa,
+    cloudAssemblyArtifact: cxa,
+    installCommand: 'do install',
+  }));
+  const hash3 = synthWithAction((sa, cxa) => cdkp.SimpleSynthAction.standardNpmSynth({
+    sourceArtifact: sa,
+    cloudAssemblyArtifact: cxa,
+    environment: {
+      computeType: cbuild.ComputeType.LARGE,
+    },
+  }));
+  const hash4 = synthWithAction((sa, cxa) => cdkp.SimpleSynthAction.standardNpmSynth({
+    sourceArtifact: sa,
+    cloudAssemblyArtifact: cxa,
+    environmentVariables: {
+      xyz: { value: 'SOME-VALUE' },
+    },
+  }));
+
+  expect(hash1).not.toEqual(hash2);
```

Comment:
`~` I wonder if it's worth having two hashes be equal as well (e.g., hash2 calculated from the exact same parameters of hash3). As-is, this test would pass if the hash was random data each time.